### PR TITLE
[Ops][Feature] Add chunk_gated_delta_rule_compute_wy operator for Ascend 310P

### PR DESCRIPTION
**Pull request by semenishinDmitry**
### What this PR does / why we need it?
This PR introduces a custom AscendC operator implementation for `ChunkGatedDeltaRuleComputeWy`, specifically targeting the Ascend 310P NPU platform. The time 0.2x from baseline.

**Why the changes are needed:**
Previously, this specific computation fallback was executed on the CPU using PyTorch. Moving this intensive operation to a dedicated NPU kernel significantly reduces execution overhead, eliminates redundant host-device synchronization/data transfers, and accelerates the overall model forward pass within the `vllm-ascend` backend.

**What changes are proposed:**
- Added the core `KernelComputeWy` class utilizing AscendC APIs for NPU vector computing.
- Integrated local WY math and chunk-based matrix operations inside the vector pipeline.

### Does this PR introduce _any_ user-facing change?
No. This is purely a backend performance optimization and platform support enhancement. The operator's external API, behavior, and output results remain identical to the existing baseline.

### How was this patch tested?
The operator was tested locally on Ascend 310P hardware:
1. **Mathematical Correctness:** Verified output tensors against the golden PyTorch CPU baseline. Results match within acceptable precision limits for half/float types.

### Performance Validation

**Setup:** Isolated prefill profiling on Ascend 310P (TP=2) using single request with 4096 input tokens and 1 output token to eliminate decode-phase noise.

**Method:** Warmup run → profiled run via `vLLM.start_profile()` → analysis with `torch_npu.analyse()` → `op_statistic.csv` for Slice op metrics.

**Key Result:** `Slice` operation optimization shows [40%] improvement in prefill latency for long-context scenarios.

Benchmark code
```
import argparse
import os

PROF_DIR = os.environ.setdefault("VLLM_TORCH_PROFILER_DIR",
                                 "/home/s60117411/profiling/prefill_slice")
os.environ.setdefault("VLLM_WORKER_MULTIPROC_METHOD", "spawn")  # NPU can't fork
os.environ.pop("ASCEND_LAUNCH_BLOCKING", None)

def main():
    ap = argparse.ArgumentParser()
    ap.add_argument("--model", default="/home/s60117411/models/Qwen/Qwen3___5-4B")
    ap.add_argument("--input-len", type=int, default=4096)
    ap.add_argument("--output-len", type=int, default=1)
    ap.add_argument("--max-model-len", type=int, default=8192)
    ap.add_argument("--gpu-mem-util", type=float, default=0.5)
    ap.add_argument("--tp", type=int, default=2)
    args = ap.parse_args()

    from vllm import LLM, SamplingParams
    from vllm.inputs import TokensPrompt

    # Создаем папку для профилей, если её нет
    os.makedirs(PROF_DIR, exist_ok=True)

    llm = LLM(
        model=args.model,
        dtype="float16",                 # 310P has no bf16
        trust_remote_code=True,
        max_model_len=args.max_model_len,
        gpu_memory_utilization=args.gpu_mem_util,
        tensor_parallel_size=args.tp,
        max_num_seqs=1,
        limit_mm_per_prompt={"image": 0, "video": 0},  # skip vision (hybrid model)
        compilation_config={"cudagraph_mode": "NONE"},  # prefill: no ACL-graph capture
        profiler_config={"profiler": "torch", "torch_profiler_dir": PROF_DIR},
    )

    sp = SamplingParams(temperature=0.0, max_tokens=args.output_len,
                        min_tokens=args.output_len, ignore_eos=True)
    prompt = TokensPrompt(prompt_token_ids=[100] * args.input_len)

    # warmup (NOT profiled) — cold kernel load / weight materialize
    print("Running warmup...")
    llm.generate([prompt], sp, use_tqdm=False)

    # profiled prefill
    print("Starting profiled generation...")
    llm.start_profile()
    outs = llm.generate([prompt], sp, use_tqdm=False)
    llm.stop_profile()

    gen = sum(len(o.outputs[0].token_ids) for o in outs)
    print(f"\n==== prefill profile done ====")
    print(f"model      : {args.model}")
    print(f"in_len={args.input_len} out_len={args.output_len} tp={args.tp}")
    print(f"gen tokens : {gen}")
    print(f"trace dir  : {PROF_DIR}")
    print("\nTo analyze:")
    print(f"  python -c \"import torch_npu; torch_npu.analyse('{PROF_DIR}')\"")
    print(f"  grep Slice {PROF_DIR}/ASCEND_PROFILER_OUTPUT/op_statistic.csv")
```

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/58d3918e3ea0a544ffedadad2ba84559e9c51d8f

### CODE REVIEW

File: csrc/moe/chunk_gated_delta_rule_compute_wy/op_kernel/arch20/compute_wy_kernel.h [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+     pipe_->InitBuffer(qHalfBuf_, chunkKElems_ * sizeof(half));
+     pipe_->InitBuffer(vHalfBuf_, chunkVElems_ * sizeof(half));
+     pipe_->InitBuffer(kFloatBuf_, chunkKElems_ * sizeof(float));
+     pipe_->InitBuffer(vFloatBuf_, chunkVElems_ * sizeof(float));
+     pipe_->InitBuffer(kBetaBuf_, chunkKElems_ * sizeof(float));
+     pipe_->InitBuffer(uAccBuf_, chunkVElems_ * sizeof(float));
+     pipe_->InitBuffer(attnBuf_, ATTEN_ELEMS * sizeof(float));
+     pipe_->InitBuffer(gBuf_, FIXED_CHUNK_SIZE * sizeof(float));
+     pipe_->InitBuffer(expGBuf_, FIXED_CHUNK_SIZE * sizeof(float));
+     pipe_->InitBuffer(tmpBuf_, (alignK_ > alignV_ ? alignK_ : alignV_) * sizeof(float));
```
[the remark this answers is not in the document]
Comment (semenishinDmitry) [reply csrc/moe/chunk_gated_delta_rule_compute_wy/op_kernel/arch20/compute_wy_kernel.h]: fixed

File: csrc/moe/chunk_gated_delta_rule_compute_wy/op_kernel/arch20/compute_wy_vector.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+class WyVectorGemm {
+ public:
+  __aicore__ inline void GemmATransB(const LocalTensor<float> &attn, const LocalTensor<float> &a,
+                                     const LocalTensor<float> &b, const LocalTensor<float> &mulTmp, uint32_t m,
+                                     uint32_t k, uint32_t lda, uint32_t ldb) const {
+    for (uint32_t i = 0; i < m; ++i) {
+      const uint32_t aRowOff = i * lda;
+      for (uint32_t j = 0; j < m; ++j) {
+        const uint32_t bRowOff = j * ldb;
+        Mul(mulTmp, a[aRowOff], b[bRowOff], k);
```
[the remark this answers is not in the document]
Comment (semenishinDmitry) [reply csrc/moe/chunk_gated_delta_rule_compute_wy/op_kernel/arch20/compute_wy_vector.h]: fixed

File: vllm_ascend/_310p/ops/fla/chunk_gated_delta_rule.py [hunk 3]
```diff
@@ -363,6 +379,62 @@ def _compute_kernel_inputs_from_torch_wy(
...
+        return False
+    if g.dtype != torch.float32:
+        return False
+    if q.ndim != 4 or k.shape != q.shape or v.ndim != 4 or g.ndim != 3 or beta.shape != g.shape:
+        return False
+    if q.shape[1] % chunk_size != 0 or v.shape[0] != q.shape[0] or v.shape[1] != q.shape[1]:
+        return False
+    if g.shape != (q.shape[0], q.shape[1], v.shape[2]):
+        return False
+    return v.shape[2] % q.shape[2] == 0
```
[the remark this answers is not in the document]
Comment (semenishinDmitry) [reply vllm_ascend/_310p/ops/fla/chunk_gated_delta_rule.py:498]: fixed

File: csrc/torch_binding.cpp [hunk 4]
```diff
@@ -2108,6 +2108,53 @@ at::Tensor chunk_fwd_o(
...
+    const int64_t B = q.size(0);
+    const int64_t T = q.size(1);
+    const int64_t Hk = q.size(2);
+    const int64_t K = q.size(3);
+    const int64_t Hv = v.size(2);
+    const int64_t V = v.size(3);
+    TORCH_CHECK(v.size(0) == B && v.size(1) == T, "v must share B/T with q.");
+    TORCH_CHECK(g.size(0) == B && g.size(1) == T && g.size(2) == Hv, "g must match [B, T, Hv].");
+    TORCH_CHECK(Hv % Hk == 0, "Hv must be divisible by Hk, got Hv=", Hv, " Hk=", Hk);
+    TORCH_CHECK(T % chunk_size_ == 0, "T must be padded to a multiple of chunk_size.");
```
[the remark this answers is not in the document]
Comment (semenishinDmitry) [reply csrc/torch_binding.cpp:2072]: fixed

**Comment by vladimirevmenoff** [comment]
[Ops][Feature] Add chunk_gated_delta_rule_compute_wy operator for Ascend 310P давай так
